build.sh

```diff
@@ -71,6 +72,23 @@ function hasArg {
     (( ${NUMARGS} != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
 }
 
+function cmakeArgs {
+    # Check for correctly formatted cmake args option
+    if [[ $(echo $ARGS | grep -E "\-\-cmake\-args=\"") ]]; then
+        CMAKE_ARGS="$(echo $ARGS | grep -Eo "\-\-cmake\-args=\".+\" ")"
+        if [[ -n ${CMAKE_ARGS} ]]; then
+	    # Remove the full  CMAKE_ARGS argument from list of args so that it passes validArgs function
+	    ARGS=${ARGS//$CMAKE_ARGS/}
+	    # Filter the full argument down to just the extra string that will be added to cmake call
+            CMAKE_ARGS=$(echo $CMAKE_ARGS | grep -Eo "\".+\"" | sed -e 's/^"//' -e 's/"$//')
+        fi
+    elif [[ $(echo $ARGS | grep -E "\-\-cmake\-args=") ]]; then
```

Comment:
Because the `build.sh` script calls `set -e` we will never hit the echo statement on line 87. Instead whenever 'invalid' cmake arguments are provided the script will silently fail.

Reply:
This only happened when --cmake-args was the last argument in the list of arguments to the script, which was very odd, but this is fixed.